pkg/network/handlers/safrole_ticket_submit.go

```diff
@@ -0,0 +1,228 @@
+func (h *SafroleTicketSubmitRequestHandler) HandleStream(ctx context.Context, stream quic.Stream, peerKey ed25519.PublicKey) error {
+	return handleSafroleTicket(ctx, stream, &h.safroleTicketHandler, true)
+}
+
+// HandleStream implements CE 132 Safrole ticket distribution protocol.
+// This method receives tickets from proxy validators during the broadcast phase.
+// All current validators should receive and store these tickets.
+//
+// Storage Process:
+// 1. Verify ticket proof is valid
+// 2. Store ticket for potential use in block sealing
+// 3. Ticket becomes available for lottery participation
+func (h *SafroleTicketDistributionRequestHandler) HandleStream(ctx context.Context, stream quic.Stream, peerKey ed25519.PublicKey) error {
+	return handleSafroleTicket(ctx, stream, &h.safroleTicketHandler, false)
+}
+
+// TODO: This will probably split into two functions, one for CE 131 and one for CE 132,
+// as the runtime becomes more complete. Currently we just save all the tickets, when complete
+// it will probably not be the case.
+// handleSafroleTicket implements the common wire protocol for both CE 131 and CE 132.
+// If requireProxyValidation is true, it checks if the current node is the correct proxy for the ticket.
+// If false, it assumes the ticket is from a proxy validator and skips the proxy check.
+// Wire Protocol (both CE 131 and CE 132):
+// --> Epoch Index ++ Ticket (Epoch index identifies when ticket will be used)
+// --> FIN
+// <-- FIN
+// Validation Steps:
+// 1. Parse epoch index and ticket proof from message
+// 2. Verify RingVRF proof against current ring commitment
+// 3. For CE 131: Check this node is correct proxy for the ticket
+// 4. Store ticket with computed hash for future use
+//
+// Security Notes:
+// - Ticket proofs are verified against the ring commitment of next epoch validators
+// - Proxy validation ensures tickets are only accepted by designated recipients
+// - Invalid tickets are rejected to prevent spam and ensure lottery integrity
+func handleSafroleTicket(ctx context.Context, stream quic.Stream, handler *safroleTicketHandler, requireProxyValidation bool) error {
+	msg, err := ReadMessageWithContext(ctx, stream)
+	if err != nil {
+		return fmt.Errorf("read request message: %w", err)
+	}
+
+	if len(msg.Content) < 4+1+block.TicketProofSize { // epoch index 4 + attempt 1 + ticket proof 784 = 789
+		return fmt.Errorf("message too short")
+	}
+
+	// Deserialize the message into structured format
+	var request safroleTicketSubmitMessage
+	if err := jam.Unmarshal(msg.Content, &request); err != nil {
+		return fmt.Errorf("unmarshal message: %w", err)
+	}
+
+	// Get the ring commitment for the next epoch's validators
+	// This is used to verify the RingVRF proof in the ticket
+	// TODO: Deal with epoch change.
+	ringCommitment, err := handler.state.ValidatorState.SafroleState.NextValidators.RingCommitment()
+	if err != nil {
+		return fmt.Errorf("get ring commitment: %w", err)
+	}
```

Comment:
```suggestion
	ringCommitment, err := handler.state.ValidatorState.SafroleState.RingCommitment
```

No need to recompute this, just take the one that's already there.

Reply:
I do, it's for next validators, not current ones.

Reply:
You are right.